source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -105,47 +105,65 @@ void MaterialXData::generateFragment(const mx::FileSearchPath& librarySearchPath
         throw mx::Exception("Invalid element to create wrapper for " + _element->getName());
     }
 
-    mx::ShaderGeneratorPtr shaderGenerator = mx::GlslFragmentGenerator::create();
-    mx::GenContext genContext(shaderGenerator);
-
-    // Set up generator context. For shaders use FIS environment lookup,
-    // but disable this for textures to avoid additional unneeded XML parameter
-    // generation.
-    genContext.registerSourceCodeSearchPath(librarySearchPath);
-    genContext.getOptions().hwSpecularEnvironmentMethod =
-        shaderRef ? mx::SPECULAR_ENVIRONMENT_FIS : mx::SPECULAR_ENVIRONMENT_NONE;
-
-    genContext.getOptions().hwMaxActiveLightSources = 0;
-    // For Maya we need to insert a V-flip fragment
-    genContext.getOptions().fileTextureVerticalFlip = true;
-
-    // Generate the fragment source (shader and XML wrapper).
-    _fragmentName = _element->getNamePath();
-    _fragmentName = mx::createValidName(_fragmentName);
-
-    _shader = shaderGenerator->generate(_fragmentName, _element, genContext);
-    if (!_shader)
     {
-        throw mx::Exception("Failed to generate shader");
+        mx::ShaderGeneratorPtr shaderGenerator = mx::GlslFragmentGenerator::create();
```

Comment:
Just limiting the scope of the shader generator and its context here, to make it clear that they the fragment generator below doesn't depend on them.